mono/utils/mono-flight-recorder.h

```diff
@@ -0,0 +1,66 @@
+/**
+ * \file
+ *     A lightweight log storage medium with limited history
+ *
+ * Author:
+ *   Alexander Kyte (alkyte@microsoft.com)
+ *
+ * (C) 2018 Microsoft, Inc.
+ *
+ */
+#ifndef __MONO_FLIGHT_RECORDER__
+#define __MONO_FLIGHT_RECORDER__
+
+#include <glib.h>
+#include <mono/utils/mono-coop-mutex.h>
+
+typedef struct {
+	long counter; // The number of messages allocated thus far, acts like a global, monotonic clock
+} MonoFlightRecorderHeader;
+
+typedef struct {
+	MonoFlightRecorderHeader header;
+	gpointer payload [MONO_ZERO_LEN_ARRAY]; // We have a variably-sized payload
+} MonoFlightRecorderItem;
+
+typedef struct {
+	intptr_t cursor; // Signed, for sentinel value of -1
+	size_t max_count; // Maximum number of items in logger
+	size_t payload_size; // Size of data reserved for logging message
```

Comment:
any reason why you don't use `gint` and `gsize` here?

Reply:
I don't see that much usage of gsize in the codebase compared to size_t (693 vs 1235 occurrences in /mono). I didn't think we should be trying to use it. I can make that change though. 

The gint change does make more sense. I used cursor as an actual pointer before, I should have changed this after I squashed my changes. Thanks for catching this.